src/runtime/markdown-parser/handler/utils.ts

```diff
@@ -20,11 +20,13 @@ export function parseThematicBlock (lang: string) {
   const language = lang.replace(/[{|[](.+)/, '').match(/^[^ \t]+(?=[ \t]|$)/)
   const highlightTokens = lang.match(/{([^}]+)}/)
   const filenameTokens = lang.match(/\[(.+)\]/)
+  const meta = lang.replace(/^\w+\s*(\[[^\]]+\])?\s*(\{[^}]+\})?\s*/g, '')
```

Comment:
Previous comment:

A problem is, that the `meta` regex has a fixed order for `lang […] {…} meta`
It is not replaced if we write `lang {…} […] meta`, even though **L21-22** allow different ordering.

Also, if we want to use `{…}` or `[…]` in the `meta` block, but not provide `highlights` or a `filename`, it will be matched (by **L21-22**).
Maybe, the user could provide an empty `[]` or `{}` block before the `meta` to prevent it from reading those parts in the meta.
Something like: `lang [] meta=['not', 'a', 'filename']` would be possible then

In my opinion, we should also `trim` the `lang` property before checking if it is empty (adding it in **L11**) and return `undefined` for the `meta` if `lang` is empty (adding in **L17**)